app/controllers/scc_accounts_controller.rb

```diff
@@ -7,6 +8,11 @@ class SccAccountsController < ApplicationController
   def index
     @scc_accounts = resource_base.search_for(params[:search], order: params[:order])
                                  .paginate(page: params[:page])
+
+    # overwrite the product list with filtered products that do not include products with empty repositories
+    @scc_accounts.each do |scc_account|
+      scc_account.scc_products = scc_account.scc_products.only_products_with_repos.order(:friendly_name)
+    end
```

Comment:
Is there a better way to access the filtered products and assign them to a specific variable?

Reply:
Isn't this "dangerous" if you assign scc_acount.scc_products to the "same" variable with scc_account.scc_products.only....?

Reply:
Intuitively, I would assume so.

Reply:
The advantage of working on the scc_account.scc_products directly is that it is already used everywhere and all changes are automatically applied to the filtered products. However, to fix this in a proper manner, some things need to be clarified:
a) What happens if a user subscribes to a product and that particular product is being removed from the repos? 
b) Do we need to keep track of products without repos (if they have never been synced before)?
c) Shall we do a sync before a subscription in case the desired product has no more valid repository since the last sync.

Reply:
Maybe it would be better to define a separate method/request, which will return all products with repositories and call that one.

The `index` page IMHO only needs the count of available products, so maybe setting that might already be enough, in order to avoid redefining `scc_account.scc_products`:
```ruby
@scc_accounts.each do |scc_account|
      scc_account.scc_product_count = scc_account.scc_products.only_products_with_repos.count
end
```
The `show`-page, which provides the available products and let's you select them requires the list of products for the account that is shown, which is fine using the helper function.

Reply:
One thing to keep in mind with the show page is that (as of now) the scc_extensions of the scc_products are set based on all the products. Maybe we could have a short discussion about the above questions and which "features" to include in the show page.